src/QsCompiler/Transformations/Attributes.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+
+
+namespace Microsoft.Quantum.QsCompiler.Transformations
+{
+    using CallablePredicate = Func<QsCallable, bool>;
+    using AttributeId = QsNullable<UserDefinedType>;
+    using QsRangeInfo = QsNullable<Tuple<QsPositionInfo, QsPositionInfo>>;
+    using AttributeSelection = IEnumerable<(QsDeclarationAttribute, Func<QsCallable, bool>)>;
+
+
+    public class Attributes
```

Comment:
Hm, the full name is Microsoft.Quantum.QsCompiler.Transformation.Attributes and correspondingly it should be a fair expectation that it is related to contain attribute related tools for transforming an existing compilation. I am not sure what name you had in mind. Do you have a concrete suggestion?

Reply:
It is hard for me to come up with a better name for this class when I am still unclear as to the use for its public static members. If you could add a description for each of those, it might help me have a better picture for what this class is for. My concern with `Attributes` is that it is fairly generic and non-descriptive about what the class is in relation to the concept of Q# Attributes.

Reply:
In lieu of better understanding of the static methods, perhaps `AttributeUtils`?

Reply:
That works.